test/shared/src/main/scala/zio/test/Assertion.scala

```diff
@@ -541,18 +650,24 @@ object Assertion extends AssertionVariants {
   val isNonEmptyString: Assertion[String] =
     Assertion.assertion("isNonEmptyString")()(_.nonEmpty)
 
-  /**
-   * Makes a new assertion that requires a None value.
-   */
-  val isNone: Assertion[Option[Any]] =
-    Assertion.assertion("isNone")()(_.isEmpty)
-
   /**
    * Makes a new assertion that requires a null value.
    */
   val isNull: Assertion[Any] =
     Assertion.assertion("isNull")()(_ == null)
 
+  /**
+   * Makes a new assertion that requires a value to be equal to one of the specified values.
+   */
+  def isOneOf[A, B](values: Set[A])(implicit eql: Eql[A, B]): Assertion[B] =
```

Comment:
Maybe this can take an `Iterable`? Not sure what the harm is if the user duplicates elements and may just be more flexible for the caller. Also would suggest removing the `eql` here as is significantly complicated the type signature and don't think it buys us additional type safety here. If you're worried about arrays we could add an additional helper method for that.

Reply:
This has been changed to accept an `Iterable`. The implicit parameter was left by mistake and has now been removed.